features/step_definitions/metering.rb

```diff
@@ -42,41 +42,52 @@
 #  ReportGenerationQueries do 'oc get reportgenerationqueries -n $METERING_NAMESPACE'
 # 2. Creating a report
 # =>
-
+# for easier to test, we default to 'runImmediately', to disable that parameter,
+# specify the parameter `run_immediately` the value `false` in the table
 Given /^I get the #{QUOTED} report and store it in the#{OPT_SYM} clipboard using:$/ do |name, cb_name, table|
   ensure_admin_tagged
 
   cb_name ||= :report
   opts = opts_array_to_hash(table.raw)
-  default_start_time = "#{Time.now.year}" + "-01-01T00:00:00Z"
-  default_end_time = "#{Time.now.year}" + "-12-30T23:59:59Z"
+  # one day before current time
+  default_start_time = (Time.now - 86400).utc.strftime('%FT%TZ') #"#{Time.now.year}" + "-01-01T00:00:00Z"
+  default_end_time = "#{Time.now.year}" + "-12-30T23:59:59Z"  # end of the year
   default_format = 'json'
+  opts[:run_immediately] ||= true
   # sanity check
   required_params = [:query_type ]
   required_params.each do |param|
     raise "Missing parameter '#{param}'" unless opts[param]
   end
+  opts[:start_time] ||= default_start_time
+  opts[:end_time] ||= default_end_time
+  opts[:run_immediately] = to_bool(opts[:run_immediately])
+  opts[:grace_period] ||= nil
+  opts[:schedule] ||= opts[:schedule]
+  opts[:format] ||= default_format
+  opts[:metadata_name] ||= name
+
+  unless opts[:use_existing_report] == 'true'
```

Comment:
can't we detect whether report already exists automatically? Or is the idea that we don't want to use report that already exists by mistake. i.e. it already exists but we do not expect it to exist.

Reply:
There two different types of reports

1. run-once report, with the parameter `runImmediately` set, which is a one and done. 
2. scheduled report which is generated from a range of time in a set interval specified by the user

The report resource created persists in the namespace.  This option is to give the user an option to reuse the object...in short, yes, we could detect the report already there.  We could change the behavior to default to delete the existing object unless the existing report is a scheduled report that has not reached the end of life.  WDYT?

Reply:
automatincally handling this makes more sense to me. I'm frustrated by too many params.

Reply:
@akostadinov there are situations in which we want to query using the same report object to save time (https://github.com/openshift/cucushift/blob/master/metering/reports.feature#L16), so I think we can leave it as is, which is if user does not specify the option `user_existing_report`, it will generate a new report object.